File: Plugin/src/main/java/com/willfp/ecoenchants/enchantments/util/EnchantmentRegisterer.java
package com.willfp.ecoenchants.enchantments.util;

import java.io.InputStream;

/**
 * Exists to simplify enchantment registration
 *
 * Prevents requiring direct class referencing on the front-end of enchantment creation
 *
 * @see com.willfp.ecoenchants.extensions.Extension
 */
public interface EnchantmentRegisterer {
    default InputStream getResourceAsStream(String location) {
        return this.getClass().getResourceAsStream(location);
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecoenchants/enchant/registration/EnchantmentRegisterer.kt
package com.willfp.ecoenchants.enchant.registration

import com.willfp.ecoenchants.enchant.EcoEnchant
import com.willfp.ecoenchants.enchant.impl.EcoEnchantBase
import org.bukkit.enchantments.Enchantment

interface EnchantmentRegisterer {
    fun register(enchant: EcoEnchantBase): Enchantment

    fun unregister(enchant: EcoEnchant)
}
